src/info/index_error.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright Redis Ltd. 2016 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "index_error.h"
+#include "rmalloc.h"
+#include "reply_macros.h"
+
+extern RedisModuleCtx *RSDummyContext;
+
+const char* NA = "N/A";
+const char* IndexError_ObjectName = "Index Errors";
+const char* IndexingFailure_String = "indexing failures";
+const char* IndexingError_String = "last indexing error";
+const char* IndexingErrorKey_String = "last indexing error key";
+
+IndexError IndexError_Init() {
+    IndexError error = {
+        .error_count = 0,           // Number of errors set to 0.
+        .last_error = (char*)NA,    // Last error message set to NA.
+        .key = RedisModule_CreateString(RSDummyContext, NA, strlen(NA)) // Key of the document that caused the error set to NA.
+    };
+    return error;
+}
+void IndexError_AddError(IndexError *error, const char *error_message, const RedisModuleString *key) {
+    if(!error_message) {
+        RedisModule_Log(RSDummyContext, "error", "Index error occurred but no index error message was set.");
+    }
+    if(error->last_error != NA) {
```

Comment:
How can we compare strings like that? shouldn't we use `strcmp`?

Reply:
We just make sure it's not the static default string. Intensionally comparing pointers here